staging/src/k8s.io/client-go/tools/events/event_recorder.go

```diff
@@ -75,16 +75,19 @@ func (recorder *recorderImpl) makeEvent(refRegarding *v1.ObjectReference, refRel
 			Name:      fmt.Sprintf("%v.%x", refRegarding.Name, t.UnixNano()),
 			Namespace: namespace,
 		},
-		EventTime:           timestamp,
-		Series:              nil,
-		ReportingController: reportingController,
-		ReportingInstance:   reportingInstance,
-		Action:              action,
-		Reason:              reason,
-		Regarding:           *refRegarding,
-		Related:             refRelated,
-		Note:                message,
-		Type:                eventtype,
+		EventTime:                timestamp,
+		Series:                   nil,
+		ReportingController:      reportingController,
+		ReportingInstance:        reportingInstance,
+		Action:                   action,
+		Reason:                   reason,
+		Regarding:                *refRegarding,
+		Related:                  refRelated,
+		Note:                     message,
+		Type:                     eventtype,
+		DeprecatedFirstTimestamp: t,
+		DeprecatedLastTimestamp:  t,
+		DeprecatedCount:          1,
```

Comment:
I'm far from being convinced this is exactly what we want.
1. We should probably change conversion to convert between deprecated_last_time <-> eventTime
  ahh - we can't do that easily because of different types, I need to think about it a bit more..
2. DeprecatedCount - we shouldn't populate it, because it will get outdated later
3. FirstTimestamp - maybe, but I need to think about it more...

Reply:
I agree mostly agree. we need to change conversion

1. we should do it anyway I think
2. maybe should be converted from EventSeries count
3. either from creationTimestamp or maybe set it until we drop these fields

is this change related to #80212 ?

Reply:
I agree mostly agree. we need to change conversion

1. we should do it anyway I think
2. maybe should be converted from EventSeries count
3. either from creationTimestamp or maybe set it until we drop these fields

is this change related to #80212 ?

Reply:
> s this change related to #80212 ?

It's not.

After thinking a bit more, I'm not even sure the conversions are broken.
@krzysied - to fix your problem, we should simply start propagating EventTime field in the old events library too:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/tools/record/event.go#L343
and switch to using EventTime field instead of FirstEventTimestamp.

Re DeprecatedLasttimestamp and DeprecatedCount, I wouldn't touch it for now.

Reply:
Done. I added EventTime initialization to corev1 and removed previous changed.